### PR TITLE
Canned Role Org admin fixes wrt recent changes

### PR DESCRIPTION
**Pull request by jyejare**
Four canned role org test cases(2 each of UI and API) are failed due to the recent changes happened in the feature. This PR fixes them:

Tests and Output after Fix:
-----------------------------------------

```
# py.test tests/foreman/ui/test_role.py -k test_positive_create_overridable_filter
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare2/Desktop/RedHat/RoboTelloNew/robottelo, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 59 items                                                                                                                                                                                                
2018-01-10 14:03:32 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_role.py .                                                                                                                                                                             [100%]

=============================================================================================== 58 tests deselected ===============================================================================================
==================================================================================== 1 passed, 58 deselected in 431.93 seconds ====================================================================================


# py.test -s -v tests/foreman/ui/test_role.py -k test_positive_create_filter_without_override
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare2/Desktop/RedHat/RoboTelloNew/robottelo, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 59 items                                                                                                                                                                                                
2018-01-10 14:07:16 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_role.py .                                                                                                                                                                             [100%]

=============================================================================================== 58 tests deselected ===============================================================================================
==================================================================================== 1 passed, 58 deselected in 292.21 seconds ====================================================================================


# py.test tests/foreman/api/test_role.py -k test_negative_create_taxonomies_by_org_admin
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare2/Desktop/RedHat/RoboTelloNew/robottelo, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 46 items                                                                                                                                                                                                
2018-01-10 14:21:00 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_role.py .                                                                                                                                                                            [100%]

=============================================================================================== 45 tests deselected ===============================================================================================
==================================================================================== 1 passed, 45 deselected in 57.96 seconds =====================================================================================


# py.test tests/foreman/api/test_role.py -k test_negative_create_roles_by_org_admin
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare2/Desktop/RedHat/RoboTelloNew/robottelo, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 46 items                                                                                                                                                                                                
2018-01-10 14:23:22 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_role.py .                                                                                                                                                                            [100%]

=============================================================================================== 45 tests deselected ===============================================================================================
==================================================================================== 1 passed, 45 deselected in 41.64 seconds =====================================================================================
```

### CODE REVIEW

File: robottelo/ui/domain.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # -*- encoding: utf-8 -*-
 """Implements Domain UI"""
+from robottelo.constants import FILTER
 from robottelo.ui.base import Base
 from robottelo.ui.locators import common_locators, locators
```
Comment (oshtaier) [inline robottelo/ui/domain.py]: that's fine to have:
`from robottelo.ui.locators import common_locators, locators, tab_locators`

File: tests/foreman/ui/test_role.py [hunk 2]
```diff
@@ -665,7 +683,10 @@ def test_positive_create_overridable_filter(self):
         with Session(self, username, password) as session:
             set_context(session, org=self.filter_org)
             set_context(session, loc=self.filter_loc)
-            make_domain(session, name=domain_name)
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: that should not be changed

File: tests/foreman/api/test_role.py [hunk 3]
```diff
@@ -1056,7 +1056,8 @@ def test_positive_taxonomies_control_to_superadmin_without_org_admin(self):
         """
 
     @tier1
-    def test_positive_create_roles_by_org_admin(self):
+    @upgrade
+    def test_negative_create_roles_by_org_admin(self):
         """Org Admin has permission to create new roles
```
Comment (rplevka) [inline tests/foreman/api/test_role.py]: Org Admin **doesn't have** the permission to create new roles

File: tests/foreman/api/test_role.py [hunk 4]
```diff
@@ -1056,7 +1056,8 @@ def test_positive_taxonomies_control_to_superadmin_without_org_admin(self):
         """
 
     @tier1
-    def test_positive_create_roles_by_org_admin(self):
```
Comment (rplevka) [inline tests/foreman/api/test_role.py]: i'm not sure whether changing the name (and a nature) of the test without changing the test ID will trigger a sync to polarion (most probably not)

Comment (jyejare) [reply tests/foreman/api/test_role.py]: Let me change that

**Pull request merged**